### ISSUE TITLE
Data Table: Add table level sort config

### ISSUE THREAD

**Issue by abhisheksinghrana** [IBM/carbon-components-angular#644]
Required Table level sort config.

Right now there is no config to enable or disable sorting for entire table.
Currently if we have to make few column sortable then first we have to make others unsortable explicitly.

**Comment by cal-smith** [comment]
So, would you say that the preferred behavior for the `table-header-item` would be to default to un-sortable?

**Comment by abhisheksinghrana** [comment]
Yes, As of now as soon as we add emitter it enable sorting for every column.
Either we keep default to non-sortable or provide any input binding in table carbon component which will be checked in component before emitting sort event.
Right now, sort.emit() is directly added on html tag in carbon component.
